core/fs/temp.test.ts

```diff
@@ -31,3 +31,47 @@ Deno.test("tempDirectory({ chdir: true }) changes working directory", async () =
   }
   assertEquals(Deno.cwd(), cwd);
 });
+
+Deno.test("tempDirectory({ chdir: true }) works recursively", async () => {
+  const originalCwd = Deno.cwd();
+  {
+    await using outerDirectory = await tempDirectory({ chdir: true });
+    const outerPath = await Deno.realPath(Deno.cwd());
+    assertEquals(outerPath, await Deno.realPath(outerDirectory.path()));
+
+    await Deno.writeTextFile("outer.txt", "outer content");
+
+    {
+      await using innerDirectory = await tempDirectory({ chdir: true });
+      const innerPath = await Deno.realPath(Deno.cwd());
+      assertEquals(innerPath, await Deno.realPath(innerDirectory.path()));
+
+      // Inner directory should be different from outer
```

Comment:
Don't add comments for what trivial code does. This was in the contribution guide.